Dockerfile.scratch

```diff
@@ -1,33 +1,49 @@
 # Build stage
+# syntax = docker/dockerfile:1.4.2@sha256:443aab4ca21183e069e7d8b2dc68006594f40bddf1b15bbd83f5137bd93e80e2
 ARG goversion
-FROM golang:${goversion}-alpine as builder
+FROM --platform=${BUILDPLATFORM} golang:${goversion}-alpine as base
 WORKDIR /spire
-RUN apk add --no-cache --update build-base musl-dev git mercurial ca-certificates
-ADD go.* ./
-RUN go mod download
-ADD . .
-RUN make build-static
+RUN apk --no-cache --update add file bash clang lld pkgconfig git make
+COPY go.* ./
+# https://go.dev/ref/mod#module-cache
+RUN --mount=type=cache,target=/go/pkg/mod go mod download
+COPY . .
 
-FROM scratch AS spire-base
+# xx is a helper for cross-compilation
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.1.2 AS xx
```

Comment:
This PR introduces a few extra build-time dependencies (qemu, regctl, xx). As a security project, we try and be mindful when these things are introduced. There probably isn't an alternative to relying on most of these. However, `xx` feels like one that we may be able to de-risk a little by copying the scripts we use and checking them into the repository. It seems unlikely the scripts would change often, except to introduce additional architectures, and this would make it easier for reviewers to audit what has changed.

We also don't want to overcomplicate things. What are your thoughts?

Reply:
We could lock the image by digest, so nobody can change it.